src/utils.ts

```diff
@@ -135,6 +137,43 @@ function words<T extends string>(sentence: T): Words<T> {
     .split(/\s+/g) as Words<T>
 }
 
+/**
+ * Truncate a string if it's longer than the given maximum length.
+ * The last characters of the truncated string are replaced with the omission string which defaults to "...".
+ */
+type Truncate<
+  T extends string,
+  Size extends number,
+  Omission extends string = '...',
+> = Math.IsNegative<Size> extends true
+  ? Omission
+  : Math.Subtract<Split<T>['length'], Size> extends 0
+  ? T
+  : `${Slice<T, 0, Math.Subtract<Size, Split<Omission>['length']>>}${Omission}`
+
+/**
+ * A strongly typed function to truncate a string .
+ * if it's longer than the given maximum string length.
+ * @param sentence the sentence to extract the words from.
+ * @param length the maximum length of the string.
+ * @param omission the string to append to the end of the truncated string.
+ * @returns the truncated string
+ * @example truncate('Hello, World', 8) // 'Hello...'
+ */
+function truncate<T extends string, S extends number, P extends string = '...'>(
+  sentence: T,
+  length: S,
+  omission = '...' as P,
+): Truncate<T, S, P> {
+  if (length <= 0) return omission as Truncate<T, S, P>
+  if (sentence.length <= length) return sentence as Truncate<T, S, P>
+  if (sentence.length <= omission.length) return omission as Truncate<T, S, P>
+  return `${sentence.slice(
+    0,
+    length - omission.length,
+  )}${omission}` as Truncate<T, S, P>
+}
+
```

Comment:
If you use the very same methods for both runtime and type implementation you might not need this type casting here.
Check the v1.1 release not where I talked about making type and runtime implementations match

```suggestion
  return join([slice(sentence,
    0,
    length - omission.length,
  ), omission])
}

```

Reply:
fixed in [f155f9b](https://github.com/gustavoguichard/string-ts/pull/41/commits/f155f9b3c532da62dd2ca2ec26686467f3e267f3)